pkg/shared/utils/utils.go

```diff
@@ -53,6 +57,39 @@ func RunCommand(cmd *exec.Cmd) ([]byte, error) {
 	return outb.Bytes(), nil
 }
 
+func RunCommandAndParseOutputLineByLine(cmd *exec.Cmd, pfn processFn) error {
+	// Get a pipe to read from standard out
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get stdout pipe: %w", err)
+	}
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get stderr pipe: %w", err)
+	}
+
+	// Start the command and check for errors
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("failed to start command: %w", err)
+	}
+
+	// Merge stdout and stderr
+	merged := io.MultiReader(stderr, stdout)
+	scanner := bufio.NewScanner(merged)
+	// Use the scanner to scan the output line by line and parse it
+	for scanner.Scan() {
+		line := scanner.Text()
+		pfn(line)
+	}
```

Comment:
We need to check scanner.Err() here to make sure we log any errors while parsing the output